src/app/pages/questions/services/questions.service.ts

```diff
@@ -98,13 +98,13 @@ export class QuestionsService {
     while (
       qIndex < groupKeys.length &&
       this.isNotNullOrEmpty(
-        groupedQuestions.get(groupKeys[qIndex]).branching_logic
+        groupedQuestions.get(groupKeys[qIndex])[0].branching_logic
       )
     ) {
       const answers = this.util.deepCopy(this.answerService.answers)
       if (
         parseAndEvalLogic(
-          groupedQuestions.get(groupKeys[qIndex]).branching_logic,
+          groupedQuestions.get(groupKeys[qIndex])[0].branching_logic,
```

Comment:
does this assume all questions in the matrix will have the same branching logic and hence only checks for the first question of the group?

Reply:
Yes, should we check for each logic in the group?

Reply:
Yes that would be ideal and i don't think there will be a huge performance penalty.

Reply:
Ok sure. If one of the logic is false, does that mean we skip the group altogether?

Reply:
No, i think it should be similar to how we handle normal questionnaires. So we only remove the question which has false branching logic and keep the other ones even within the same matrix. This will be consistent with other questionnaire types and also with redcap. If it is too much work, maybe create an issue and can be handled later, till then we can just use matrix level branching logic (which you have currently)

Reply:
I've made the changes. It will now filter out the questions whose logic return false.